### PR TITLE
IDea: footer support like was with components

### PR DESCRIPTION
**Pull request by mckaygerhard**
as same components did with navbar i noted same can be done with few lines with footer, thsi commit does not include the footer.md file

### CODE REVIEW

**Comment by arnaudsm** [comment]
Great addition thank you ! 
Would you mind adding an example `docs/footer.md` since you added it to the config ?

**Comment by mckaygerhard** [comment]
yeah i can add in this PR the example , just gime me some hours that i have to deal with some home task XD @arnaudsm

**Comment by mckaygerhard** [comment]
> Great addition thank you ! Would you mind adding an example `docs/footer.md` since you added it to the config ?

Done with commit 280b46452a11748e48a879311cb1c534c3c45157 @arnaudsm  i hope i do it in right way cos i made it using the phone (was so uncomfortable )

File: docs/footer.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+-----------------------------------------------------------------------------------
+
+This was made with love by 💼 [https://arnaud.at](https://arnaud.at) under MIT License
```
Comment (arnaudsm) [inline docs/footer.md]:
```suggestion
Made with love by [arnaudsm](https://arnaud.at) under MIT License
```

File: index.html [hunk 2]
```diff
@@ -301,6 +302,7 @@
 <body>
```
Comment (arnaudsm) [inline index.html:314]: I suggest to add the following style 
```css
	#docs-footer {
		text-align: center;
		background-color: var(--color-1);
		color: white;
		padding: 1rem;
	}

	#docs-footer a {
		color: white;
		text-decoration: underline;
	}
```

image

File: docs/README.md [hunk 3]
```diff
@@ -26,21 +26,27 @@ CloudFlare Pages does it by default.
 | `browserRouter: true`  | `example.com/page`   | ✅            |
 
 
-## Components
+## Components and footer
+
 Components are visible in every page. Useful for navbars, sidebars and footers.
 
-Create your component in a `.md` file, then add it to the `components` list in `index.html`.
+Create your `.md` file under the structure:
```
Comment (arnaudsm) [inline docs/README.md]:
```suggestion
Create your component in a `.md` file, then add it to the `components` or `footers` list in `index.html`.

Components are added before the content div, and Footers after it.
```

**Review by arnaudsm — COMMENTED** [review]
One last review and it should be good to go!

File: index.html (hunk 2 above)
Comment (mckaygerhard) [reply index.html:314]: i want to congratulate you for the simplicity, each style is simply the name of the file with its path... wonderful @arnaudsm

Comment (mckaygerhard) [reply index.html:314]: done with commit https://github.com/arnaudsm/raito/pull/48/commits/6a5204846eb0380ecd3ea04bed595ea1256b0798

**Comment by arnaudsm** [comment]
One last thing, the feature broke the Subdirectory test.

Replace `getByText("b")` by `getByText("b", { exact: true })` in `main.test.js`

You can run the tests with
```bash
cd tests
yarn
npx playwright install chromium
yarn test # or `yarn debug` for interactive view
```

File: tests/main.test.js [hunk 4]
```diff
@@ -69,14 +69,14 @@ test("Subdirectories", async ({ page, prefix, baseUrl }) => {
 	await expect(await page.getByText("Docs").getAttribute("href")).toEqual(prefix + "/docs/");
 	await expect(await page.getByText("Homepage").getAttribute("href")).toEqual(prefix || "/");
 
-	await page.getByText("b").click();
+	await page.getByText("b", { exact: true }).click();
```
Comment (mckaygerhard) [inline tests/main.test.js:72]: i can understand why this change.. will query inside [baseElement](https://testing-library.com/docs/react-testing-library/api#baseelement) but you use `page` objest so exact: true i guess is a way to dont choose another different thant the exact href to test? i dont know about such parameter for

Comment (mckaygerhard) [inline tests/main.test.js:72]: i cannot understand why this change.. will query inside [baseElement](https://testing-library.com/docs/react-testing-library/api#baseelement) but you use `page` object so [exact: true](https://testing-library.com/docs/queries/bytext/) i guess is a way to dont choose another different than the exact href to test? i dont know about such parameter for, the "`b`" is too ambiguous or generic but "`page`" must be enough to difference it

Comment (arnaudsm) [reply tests/main.test.js:72]: Previously, we matched the character B anywhere in the page, which is too permissive, and why it broke in this PR.
The proposed fix is to make the text match exact, which means the anchor should exactly be equal to B.

**Review by arnaudsm — APPROVED** [review]
Thanks, that feature is a great addition 👍
You can merge it

**Comment by mckaygerhard** [comment]
> Thanks, that feature is a great addition 👍 You can merge it

err .. men! i m just a collaborator XD XD XD i cannot merge it XD

**Pull request merged**